components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -674,8 +674,10 @@ class GeckoEngineSession(
 
             if (interceptionResponse !is InterceptionResponse.AppIntent) {
                 appRedirectUrl = ""
-                lastLoadRequestUri = request.uri
             }
+
+            lastLoadRequestUri = request.uri
```

Comment:
I suggest not to remove line 677.
We then create an issue to fix the root cause.  But for now we add this workaround for Focus.  Please let me know if this is a good compromise. 

```suggestion
            lastLoadRequestUri = request.uri
        }
            
        // TODO fix root cause: https://github.com/mozilla-mobile/android-components/issues/<TBD>
        if (interceptor != null && interceptor.interceptsAppInitiatedRequests()) {
            lastLoadRequestUri = request.uri
        }
```

Reply:
Done